mlrun/api/utils/builder.py

```diff
@@ -236,7 +238,10 @@ def make_kaniko_pod(
         default_pod_spec_attributes=extra_runtime_spec,
         resources=resources,
     )
-    kpod.env = builder_env
+    envs = (
+        builder_env if builder_env else [] + project_secrets if project_secrets else []
```

Comment:
Why do you ignore the `project_secrets` if there is a `builder_env`? Don't you need to add the `project_secrets` anyway? Otherwise if I specify a `builder_env` then I can't pass project-secrets as builder args. Am I missing something?

Reply:
What it does is screaming a list of builder_env + project_secrets, and it will be None only if both are None

Reply:
Oh, it's the terrible Python notation. To make sure I understand, this means -
```
(builder_env if builder_env else []) + project_secrets
```
I read it as:
```
(builder_env if builder_env) else (([] + project_secrets) if project_secrets else [])
```
I hate this notation, it's terribly confusing.

Reply:
Yes 🙃
Liran said the exact same thing, so I splited it into two lines in the merged version, I hope it is more clear, maybe I'll add a comment there as well